### PR TITLE
Set min-cpu-platform to haswell to support avx2 in deployment manager

### PR DESCRIPTION
**Pull request by lluunn**
For #1082 
/hold

/cc @kunmingg

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1083)

### CODE REVIEW

**Comment by lluunn** [comment]
Seeing error now:
```
ERROR: (gcloud.deployment-manager.deployments.create) Error in Operation [operation-1530066747939-56f966cfe58b8-266cb5d7-c8c932c1]: errors:
- code: RESOURCE_ERROR
  location: /deployments/kubeflow1/resources/kubeflow1-cpu-pool-v1
  message: '{"ResourceType":"container.v1.nodePool","ResourceErrorCode":"400","ResourceErrorMessage":{"code":400,"message":"Cluster
    is currently being created, deleted, updated or repaired and cannot be updated.","status":"FAILED_PRECONDITION","statusMessage":"Bad
    Request","requestPath":"https://container.googleapis.com/v1/projects/kai-test2/zones/us-central1-a/clusters/kubeflow1/nodePools","httpMethod":"POST"}}'
```

**Comment by jlewi** [comment]
@lluunn Ping?

**Comment by lluunn** [comment]
Should we just document it on how to solve avx2 issue and close this?
If we set minCpu as this PR, some region will fail.

**Comment by jlewi** [comment]
@lluunn Yes I think we should just document the fact that we require regions with haswell to support TF

File: scripts/gke/deployment_manager_configs/cluster.jinja [hunk 1]
```diff
@@ -111,6 +111,7 @@ resources:
           machineType: n1-standard-8
           serviceAccount: {{ KF_VM_SA_NAME }}@{{ env['project'] }}.iam.gserviceaccount.com
           oauthScopes: {{ VM_OAUTH_SCOPES }}
+          minCpuPlatform: 'Intel Haswell'
```
Comment (jlewi) [inline scripts/gke/deployment_manager_configs/cluster.jinja]: Can we add a comment explaining why we require this?

Comment (lluunn) [reply scripts/gke/deployment_manager_configs/cluster.jinja]: Done

**Pull request merged**